.azure/pipelines/ci.yml

```diff
@@ -701,6 +701,40 @@ extends:
               publishOnError: true
               includeForks: true
 
+        # Local development validation
+        - template: .azure/pipelines/jobs/default-build.yml@self
```

Comment:
I'd recommend adding a condition so this only runs on PRs, similar to Code Check: https://github.com/dotnet/aspnetcore/blob/63c492e22b06d4903cb4e7aee037295c71e1ec37/.azure/pipelines/ci.yml#L158-L159